### PR TITLE
refactor: PaasNs controller

### PR DESCRIPTION
**Pull request by portly-halicore-76**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Relates to: https://github.com/belastingdienst/opr-paas/issues/155

While working on this feature, I felt the need to refactor a bit of the logic in the PaasNs controller. To scope the PR, I've decided to split them into two PR's.

## What is the new behavior?

* Reorder steps in r.GetPaasNs to match kubebuilders examples which follow logical order IMHO
* Clean-up commented code
* Can't retrieve an object without a Name, 
* Use ignoreNotFound
* Use logLevel info where is more appropriate
* Log less, some lines where duplicated

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/controller/paasns_controller.go [hunk 1]
```diff
@@ -293,13 +282,7 @@ func (r *PaasNSReconciler) paasFromPaasNs(ctx context.Context, paasns *v1alpha1.
 	logger := getLogger(ctx, paasns, "PaasNs", "paasFromPaasNs")
 	paas = &v1alpha1.Paas{}
 	if err := r.Get(ctx, types.NamespacedName{Name: paasns.Spec.Paas}, paas); err != nil {
-		err = fmt.Errorf("cannot find PaaS %s", paasns.Spec.Paas)
-		paasns.Status.AddMessage(v1alpha1.PaasStatusError, v1alpha1.PaasStatusFind, paasns, err.Error())
-		logger.Error(err, fmt.Sprintf("Cannot find PaaS %s", paasns.Spec.Paas))
-		return nil, namespaces, err
-	} else if paas.Name == "" {
```
Comment (portly-halicore-76) [inline internal/controller/paasns_controller.go:300]: Can't get an object returned without a name so this check has no value.

File: internal/controller/paasns_controller.go [hunk 2]
```diff
@@ -49,37 +49,39 @@ func (pr PaasNSReconciler) GetScheme() *runtime.Scheme {
...
 func (r *PaasNSReconciler) GetPaasNs(ctx context.Context, req ctrl.Request) (paasns *v1alpha1.PaasNS, err error) {
-	paasns = &v1alpha1.PaasNS{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: req.Name,
-		},
-	}
+	paasns = &v1alpha1.PaasNS{}
 	logger := getLogger(ctx, paasns, paasns.Kind, req.Name)
-	logger.Info("Reconciling the PaasNs object")
+	logger.Info("Reconciling PaasNs")
```
Comment (hikarukin) [inline internal/controller/paasns_controller.go:54]: lower case? 😄

File: internal/controller/paasns_controller.go [hunk 3]
```diff
@@ -49,37 +49,39 @@ func (pr PaasNSReconciler) GetScheme() *runtime.Scheme {
...
 
 	if err = r.Get(ctx, req.NamespacedName, paasns); err != nil {
-		if errors.IsNotFound(err) {
-			// Something fishy is going on
-			// Maybe someone cleaned the finalizers and then removed the PaasNs resource?
-			logger.Info(req.NamespacedName.Name + " is already gone")
-			// return ctrl.Result{}, fmt.Errorf("PaasNs object %s already gone", req.NamespacedName)
-			return nil, nil
+		return nil, client.IgnoreNotFound(err)
```
Comment (hikarukin) [inline internal/controller/paasns_controller.go:57]: Are we sure we don't want to know (informational) that there was an apparent inconsistency between configured state and reality?

File: internal/controller/paasns_controller.go [hunk 4]
```diff
@@ -49,37 +49,39 @@ func (pr PaasNSReconciler) GetScheme() *runtime.Scheme {
...
+
+	// Add finalizer for this CR
+	if !controllerutil.ContainsFinalizer(paasns, paasNsFinalizer) {
+		if ok := controllerutil.AddFinalizer(paasns, paasNsFinalizer); !ok {
+			return nil, fmt.Errorf("failed to add finalizer")
 		}
-		return nil, err
-	} else if paasns.GetDeletionTimestamp() != nil {
+		if err := r.Update(ctx, paasns); err != nil {
+			logger.Error(err, "Error updating PaasNs")
```
Comment (hikarukin) [inline internal/controller/paasns_controller.go:66]: LC, mind yer consistency

**Review by hikarukin — COMMENTED** [review]
Looking good, just a couple of nitpicks

File: internal/controller/paasns_controller.go (hunk 4 above)
Comment (portly-halicore-76) [reply internal/controller/paasns_controller.go:66]: Will be done by merging #157

File: internal/controller/paasns_controller.go (hunk 2 above)
Comment (portly-halicore-76) [reply internal/controller/paasns_controller.go:54]: Will be done by merging #157

**Review by hikarukin — APPROVED** [review]

File: internal/controller/paasns_controller.go (hunk 3 above)
Comment (portly-halicore-76) [reply internal/controller/paasns_controller.go:57]: Nah, this happens after the object has been finalized and thus removed by k8s. The reconcile after that, falls into the catagory of a reconciliation for an object which cannot be find. There can't be another case. I'm personally not interested in logging those requests.

Comment (hikarukin) [reply internal/controller/paasns_controller.go:57]: OK

**Pull request merged**